src/Umbraco.Forms.Integrations.Commerce.EMerchantPay/Controllers/PaymentProviderController.cs

```diff
@@ -71,9 +75,15 @@ public HttpResponseMessage NotifyPayment(string formId, string recordUniqueId, s
 #endif
 
                 var record = _recordStorage.GetRecordByUniqueId(Guid.Parse(recordUniqueId), form);
+
                 var paymentStatusField = record.GetRecordField(Guid.Parse(statusFieldId));
                 paymentStatusField.Values.Add(reconcileResponse.Status);
 
+                if(approve)
```

Comment:
This should only happen if the payment was successful.  Is this the case here?  In other words, do we only get a call to the `NotificationUrl` when the payment is successful, or does it happen on payment failed too?

If the former, we're fine.  If not, we need to determine the status here before approving.

Also, if we do get a callback here for all payment states, it might be worth moving all the record updates here (so also update the fields of the record too).  Then all the updates are done in one place.

Reply:
A notification is sent to our API every time a payment action occurs in the WPF form. So on approve, cancel or fail.
Added clause to execute when payment is approved.

Reply:
Looks good.  Just also thinking:

- In this line - `paymentStatusField.Values.Add(reconcileResponse.Status);`- are you going to add a second value here when really you should be updating the one that's there?  So the value goes from "pending" to "approved" (or similar) rather than "pending, approved".
- You should probably call `_recordStorage.UpdateRecord()` before `_recordService.Approve()`.  Just so any workflows attached to "approve" will have the full details of the record.

Reply:
Makes sense, made the proper amends.